appserver/batch/glassfish-batch-connector/src/main/java/fish/payara/jbatch/persistence/rdbms/DB2PersistenceManager.java

```diff
@@ -201,7 +201,7 @@ protected void createDB2TableNotExists(String tableName,
 			stmt = conn.createStatement(ResultSet.TYPE_SCROLL_INSENSITIVE,
 					ResultSet.CONCUR_READ_ONLY);
 			String query = "select name from sysibm.systables where name ="
-					+ "\'" + tableName + "\'" + "and type = 'T'";
+					+ "\'" + tableName.toUpperCase() + "\'" + "and type = 'T'";
```

Comment:
I have learnt toUpperCase() is not completely risk-free. This test is failing on e.g. Turkish locale:
```java
@Test
    public void testLocale() {
        String original = "i";
        String expectedOutcome = "I";

        assertEquals(expectedOutcome, original.toUpperCase());
    }
```

```
org.junit.ComparisonFailure: 
Expected :I
Actual   :İ
```

Some languages have weird rules :) I'm not sure what's best solution. Perhaps you deserve it when you are using something else than en_US on your server :)))